src/hooks/useGetAllReleases.ts

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { PipelineRunLabel } from '../consts/pipelinerun';
+import { K8sQueryReleaseListItems } from '../k8s';
+import { ReleaseModel } from '../models';
+
+export interface Release {
+  releasePlan: string;
+  application: string;
+  tenant: string;
+  name: string;
+  completionTime: string;
+  status: string;
+  component: string;
+}
+
+/**
+ * Get all the valid releases accorss the namespaces the you can see.
+ *
+ * @returns the array of the Release interface.
+ */
+export const useGetAllReleases = (loaded, namespaces) => {
+  const [releases, setData] = React.useState<Release[]>([]);
+
+  React.useEffect(() => {
+    const fetchAll = async (): Promise<Release[]> => {
+      const results: Release[] = [];
+      const namespacePromises = namespaces.map(async (namespace) => {
+        const nsReleases = await K8sQueryReleaseListItems({
+          model: ReleaseModel,
+          queryOptions: { ns: namespace.metadata.name },
+        });
+        nsReleases.forEach((re) => {
+          const applicationName = re.metadata?.labels?.[PipelineRunLabel.APPLICATION] ?? '';
+          const componentName = re.metadata?.labels?.[PipelineRunLabel.COMPONENT] ?? '';
+          const completionTime = re.status?.completionTime ?? '';
```

Comment:
how about we do something like ‘logic separation’, then we could likely have something like transformReleaseItem(re, namespace)
then the long function would be short and clear.

Reply:
Can you share an example?